### PR TITLE
Add new rancher setting patch in e2e

### PR DESCRIPTION
**Pull request by alexander-demicev**
**What this PR does / why we need it**:

For testing purposes we need to set agent-tls-mode to system-store. More details on the setting here: https://github.com/rancher/rancher/pull/45684

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Comment by furkatgofurov7** [comment]
@alexander-demicev I checked new setting was introduced in rancher-2.9? And we recently bumped it in turtles e2e https://github.com/rancher/turtles/pull/658. Should we re-trigger e2e tests once again to test it?

**Comment by alexander-demicev** [comment]
@furkatgofurov7 I rebased the PR

**Review by Danil-Grigorev — APPROVED** [review]

**Review by furkatgofurov7 — APPROVED** [review]
Deferring on CI, but this looks good

**Comment by Danil-Grigorev** [comment]
Can you elaborate more why is this setting needed?

**Comment by furkatgofurov7** [comment]
> Can you elaborate more why is this setting needed?

@Danil-Grigorev https://github.com/rancher/rancher/issues/45628#issuecomment-2246152604 should give an idea

File: test/e2e/data/rancher/rancher-setting-patch.yaml [hunk 1]
```diff
@@ -15,3 +15,9 @@ kind: Setting
 metadata:
   name: eula-agreed
 value: "2023-08-16T20:39:11.062Z"
+---
+apiVersion: management.cattle.io/v3
+kind: Setting
+metadata:
+  name: agent-tls-mode
+value: "system-store"
```
Comment (furkatgofurov7) [inline test/e2e/data/rancher/rancher-setting-patch.yaml:23]: By reading the linked issue and PR again, it seems we need to set this to "strict" if using Rancher 2.9

**Comment by Danil-Grigorev** [comment]
@furkatgofurov7 but this does not explain why we test this setting in our suite. Since this has to be enabled with turtles on helm chart installation in this case.

File: test/e2e/data/rancher/rancher-setting-patch.yaml (hunk 1 above)
Comment (Danil-Grigorev) [reply test/e2e/data/rancher/rancher-setting-patch.yaml:23]: `system-store` will surface the error on the cluster, which is better for debugging purposes.

**Comment by alexander-demicev** [comment]
@Danil-Grigorev  Defaulting to "strict" is a breaking change for anyone using ngrok in the way we're used to, when enabled allows only certificates verified with CAcerts authority provided by `cacerts` rancher setting will be used.
This is something that Rancher users have to configure on their own and shouldn't be related to how turtles import clusters. I believe that by using `system-store` we are not testing new functionality and using the setup we had before.
Here are a couple links to related PRs:
https://github.com/rancher/fleet/pull/2507
https://github.com/rancher/rancher/pull/45909

I'm not against extending the E2E suite to use "strict" mode for all our suites but it can done later separately.

**Comment by Danil-Grigorev** [comment]
I'm confused @alexander-demicev as I was not advocating for using strict mode, see https://github.com/rancher/turtles/pull/660#discussion_r1718214400

So if you saying that strict became new default, and by applying it we re-enables use of ngrok, change makes sense to me.

(Maybe this needs to be mentioned in PR description?)

**Comment by alexander-demicev** [comment]
@Danil-Grigorev We will need to provide CA public key for ngrok(or anything else) if using strict, which shouldn't be a problem for us or users but it can be addressed separately

**Pull request merged**